osu.Game.Rulesets.Osu/Edit/Blueprints/Sliders/Components/PathControlPointVisualiser.cs

```diff
@@ -437,7 +446,7 @@ public void DragInProgress(DragEvent e)
                 Vector2 newHeadPosition = Parent!.ToScreenSpace(e.MousePosition + (dragStartPositions[0] - dragStartPositions[draggedControlPointIndex]));
 
                 var result = positionSnapProvider?.TrySnapToNearbyObjects(newHeadPosition, oldStartTime);
-                result ??= positionSnapProvider?.TrySnapToDistanceGrid(newHeadPosition);
+                result ??= positionSnapProvider?.TrySnapToDistanceGrid(newHeadPosition, limitedDistanceSnap.Value ? oldStartTime : null);
```

Comment:
I kind of wonder if this case even makes sense. To elaborate, this is the case where you're dragging a set of slider path control points that includes the slider head control point. I debated whether applying limited distance snap even made sense in this instance, and if it did, to which time snapping should be performed.

This is partially why I tore down all the abstractions earlier in https://github.com/ppy/osu/pull/31655 - they really cover a lot of these head-scratchers up.

Reply:
I'm not sure either, let's wait for someone to complain I guess.